src/Files.App/Views/HomePage.xaml.cs

```diff
@@ -82,20 +83,23 @@ protected override void OnNavigatedFrom(NavigationEventArgs e)
 			Dispose();
 		}
 
-		private async void ToolbarViewModel_RefreshRequested(object? sender, EventArgs e)
+		private async void MultitaskingControl_CurrentInstanceChanged(object? sender, CurrentInstanceChangedEventArgs e)
```

Comment:
Does this fire when switching tabs?

Reply:
I haven't tested, but it looks like we can put this code under `OnNavigatedTo`.

Reply:
I tried it and it doesn't work. `OnNavigatedTo` is executed only when it's the first time you open the page

Reply:
What happens if you turn off caching for the page?

Reply:
> Does this fire when switching tabs?

Yes it is fired

Reply:
> What happens if you turn off caching for the page?

Just checked and it's already off, so it doesn't work

Reply:
Ideally it would only update when navigating or refreshing.

Reply:
> I tried it and it doesn't work. OnNavigatedTo is executed only when it's the first time you open the page

It appears to fire when I renavigate as well.

Reply:
I'll try to figure out why mine behaves differently

Reply:
I think we are talking about two different things:
- I'm saying it doesn't fire when you switch tab
- You are saying it fires when you navigate back/forward (which works for me too), am I right?

Reply:
Currently, this will reload the home page whenever a user switches between tabs. The expected behavior is for it to reload only if the user navigates to the home page or refreshes manually.

Reply:
In this case I can remove the call since widgets are already reloaded on navigation